src/openms_gui/source/VISUAL/SpectraIDViewTab.cpp

```diff
@@ -135,25 +184,242 @@ namespace OpenMS
   void SpectraIDViewTab::clear()
   {
     table_widget_->clear();
+    protein_table_widget_->clear();
     layer_ = nullptr;
   }
 
+  // Create the protein accession to peptide identification map using C++ STL unordered_map
+  void SpectraIDViewTab::createProteinToPeptideIDMap_()
+  {
+    //clear the map each time entries are updated with updateEntries()
+    protein_to_peptide_id_map.clear();
+
+    if (is_first_time_loading)
+    {
+      for (const auto& spec : *layer_->getPeakData())
+      {
+        if (!spec.getPeptideIdentifications().empty())
+        {
+          const vector<PeptideIdentification>& peptide_ids = spec.getPeptideIdentifications();
+
+          for (const auto& pepid : peptide_ids)
+          {
+            const vector<PeptideHit>& pep_hits = pepid.getHits();
+            //add id_accession as the key of the map and push the peptideID to the vector value-
+            for (const auto & pep_hit : pep_hits)
+            {
+              const vector<PeptideEvidence>& evidences = pep_hit.getPeptideEvidences();
+
+              for (const auto & evidence : evidences)
+              {
+                const String& id_accession = evidence.getProteinAccession();
+                protein_to_peptide_id_map[id_accession].push_back(&pepid);
+              }
+            }
+          }
+        }
+      }
+      // set is_first_time_loading to false so that the map gets created only the first time!
+      is_first_time_loading = false;
```

Comment:
The initial plan was to create the map only if we select a new file, but the updateEntries() function(which calls this function to create the map) gets called every time there is a focus change. So in future, if we fix that issue or add another method, this might come in handy. Right now this does not serve any purpose